app/helpers/maintenance_tasks/tasks_helper.rb

```diff
@@ -118,6 +132,13 @@ def resolve_inclusion_value(task_class, parameter_name)
       return unless inclusion_validator
 
       in_option = inclusion_validator.options[:in] || inclusion_validator.options[:within]
+      in_option = in_option.call if in_option.is_a?(Proc) && in_option.arity.zero?
+
+      if in_option.is_a?(Enumerable)
+        items = in_option.take(MAX_INCLUSION_ITEMS + 1).to_a
```

Comment:
nit, un-necessary as `.take` already returns an Array:

```suggestion
        items = in_option.take(MAX_INCLUSION_ITEMS + 1)
```

Reply:
Take doesn't return an array if `in_option` is a lazy enumerator:
```rb
(1..).lazy.take(10)
=> #<Enumerator::Lazy: ...>
```